README
NAME
    Dist::Zilla::PluginBundle::Git::CheckFor - All Git::CheckFor plugins at
    once

VERSION
    This document describes version 0.006 of
    Dist::Zilla::PluginBundle::Git::CheckFor - released November 14, 2012 as
    part of Dist-Zilla-PluginBundle-Git-CheckFor.

SYNOPSIS
        ; in dist.ini
        [@Git::CheckFor]

DESCRIPTION
    This bundles several plugins that do some sanity/lint checking of your
    git repository; namely: you're on the right branch and you haven't
    forgotten any autosquash commits ("fixup!" or "squash!").

SEE ALSO
    Please see those modules/websites for more information related to this
    module.

    *   Dist::Zilla::Plugin::Git::CheckFor::Fixups

    *   Dist::Zilla::Plugin::Git::CheckFor::CorrectBranch

    *   Dist::Zilla::Plugin::Git::CheckFor::MergeConflicts

    *   Dist::Zilla::PluginBundle::Git

SOURCE
    The development version is on github at
    <http://github.com/RsrchBoy/dist-zilla-pluginbundle-git-checkfor> and
    may be cloned from
    <git://github.com/RsrchBoy/dist-zilla-pluginbundle-git-checkfor.git>

BUGS
    Please report any bugs or feature requests on the bugtracker website
    https://github.com/RsrchBoy/dist-zilla-pluginbundle-git-checkfor/issues

    When submitting a bug or request, please include a test-file or a patch
    to an existing test-file that illustrates the bug or desired feature.

AUTHOR
    Chris Weyl <cweyl@alumni.drew.edu>

CONTRIBUTORS
    *   Karen Etheridge ETHER <https://metacpan.org/author/ETHER>

    *   Mike Doherty DOHERTY <https://metacpan.org/author/DOHERTY>

COPYRIGHT AND LICENSE
    This software is Copyright (c) 2012 by Chris Weyl.

    This is free software, licensed under:

      The GNU Lesser General Public License, Version 2.1, February 1999



Changes
Revision history for Dist-Zilla-PluginBundle-Git-CheckFor

0.006     2012-11-14 22:14:15 PST8PDT
	* Add plugin to check for common merge mishaps [Mike Doherty]
	* Add explicit dependency on the git executable

0.005     2012-10-03 22:10:53 PST8PDT
	* make our test suite usable on !unix platforms (github #4, #5)
	* Fix breakage introduced by the git NextVersion plugin in  1.122770 by
	  pulling over the removed code and integrating it here.  This is expected
	  to be a temporary measure as we figure out a more permanent (and
	  hopefully cooperative) solution.

0.004     2012-05-04 14:53:44 America/Los_Angeles
	* [Git::CheckFor::CorrectBranch] now permits releases from multiple brnaches
	  (Karen Etheridge)

0.003     2012-05-02 11:11:39 America/Los_Angeles
	* handle version tags where '-TRIAL' is appeneded.  We'll need to revisit this
	  later to make things Work Better, but for now it works and with a minimum of
	  fuss.

0.002     2012-04-13 14:16:08 America/Los_Angeles
	* no functional changes!
	* bump Test::Moose::More prereq to 0.008, as we need a version of
	  has_attribute_ok that handles checking roles w/o barfing (as Test::Moose now
	  seems to)

0.001     2012-02-06 13:11:50 America/Los_Angeles


t/funcs.pm
#
# This file is part of Dist-Zilla-PluginBundle-Git-CheckFor
#
# This software is Copyright (c) 2012 by Chris Weyl.
#
# This is free software, licensed under:
#
#   The GNU Lesser General Public License, Version 2.1, February 1999
#
use strict;
use warnings;

use autodie 'system';
use IPC::System::Simple (); # for autodie && prereqs

use File::chdir;
use Path::Class;

use Test::DZil;

sub make_test_repo {
    my @commands = @_;

    my $tempdir = tempdir;
    my $repo_root = dir($tempdir)->absolute;
    note "Repo being created at $repo_root";
    local $CWD = "$repo_root";

    unshift @commands, 'git init'
        unless $commands[0] =~ /git init/;

    system "($_) 2> /dev/null > /dev/null" for @commands;

    return $repo_root;
}

sub _ack {
    my ($fn, $text, $msg) = @_;
    $text ||= 'whee';
    $msg  ||= 'ack';

    return (
        qq{echo "$text" >> $fn},
        qq{git add $fn && git commit -m '$msg'},
    );
}


sub prep_for_testing {
    # repo_init, plugin_list
    my %opts = @_;

    # make some basic commits, branch...
    my ($repo_root) = make_test_repo(@{$opts{repo_init}});

    my @dist_ini = @{$opts{plugin_list}};
    unshift @dist_ini, $opts{core_args} if $opts{core_args};

    # ...then create a Builder and check for exception
    my $tzil = Builder->from_config(
        { dist_root => "$repo_root" },
        {
            add_files => {
                'source/dist.ini' => simple_ini(@dist_ini),
            },
        },
    );

    return ($tzil, $repo_root);
}

# blatantly stolen from Dist-Zilla-Plugin-CheckPrereqsIndexed-0.008/t/basic.t
# Write the log messages as diagnostics:
sub diag_log
{
  my $tzil = shift;

  # Output nothing if all tests passed:
  my $all_passed = shift;
  $all_passed &&= $_ for @_;

  return if $all_passed;

  diag(map { "$_\n" } @{ $tzil->log_messages });
}

!!42;


Makefile.PL

use strict;
use warnings;

use 5.006;

use ExtUtils::MakeMaker 6.30;



my %WriteMakefileArgs = (
  "ABSTRACT" => "All Git::CheckFor plugins at once",
  "AUTHOR" => "Chris Weyl <cweyl\@alumni.drew.edu>",
  "BUILD_REQUIRES" => {
    "Capture::Tiny" => 0,
    "Env::Path" => 0,
    "File::Find" => 0,
    "File::Temp" => 0,
    "File::chdir" => 0,
    "Path::Class" => 0,
    "Test::DZil" => 0,
    "Test::Fatal" => 0,
    "Test::Moose::More" => "0.008",
    "Test::More" => "0.88",
    "Test::Most" => 0,
    "Test::TempDir" => 0
  },
  "CONFIGURE_REQUIRES" => {
    "ExtUtils::MakeMaker" => "6.30"
  },
  "DISTNAME" => "Dist-Zilla-PluginBundle-Git-CheckFor",
  "EXE_FILES" => [],
  "LICENSE" => "lgpl",
  "NAME" => "Dist::Zilla::PluginBundle::Git::CheckFor",
  "PREREQ_PM" => {
    "Dist::Zilla::Plugin::Git::NextVersion" => "1.120370",
    "Dist::Zilla::Role::BeforeRelease" => 0,
    "Dist::Zilla::Role::Git::Repo" => 0,
    "Dist::Zilla::Role::PluginBundle::Easy" => 0,
    "Git::Wrapper" => 0,
    "IPC::System::Simple" => 0,
    "List::MoreUtils" => 0,
    "List::Util" => 0,
    "Moose" => 0,
    "Moose::Autobox" => 0,
    "Moose::Role" => 0,
    "MooseX::AttributeShortcuts" => 0,
    "Try::Tiny" => 0,
    "autodie" => 0,
    "namespace::autoclean" => 0,
    "strict" => 0,
    "warnings" => 0
  },
  "VERSION" => "0.006",
  "test" => {
    "TESTS" => "t/*.t t/plugin/*.t"
  }
);


unless ( eval { ExtUtils::MakeMaker->VERSION(6.56) } ) {
  my $br = delete $WriteMakefileArgs{BUILD_REQUIRES};
  my $pp = $WriteMakefileArgs{PREREQ_PM};
  for my $mod ( keys %$br ) {
    if ( exists $pp->{$mod} ) {
      $pp->{$mod} = $br->{$mod} if $br->{$mod} > $pp->{$mod};
    }
    else {
      $pp->{$mod} = $br->{$mod};
    }
  }
}

delete $WriteMakefileArgs{CONFIGURE_REQUIRES}
  unless eval { ExtUtils::MakeMaker->VERSION(6.52) };

WriteMakefile(%WriteMakefileArgs);



